src/overlay/SurveyManager.cpp

```diff
@@ -769,17 +788,13 @@ SurveyManager::addPeerToBacklog(NodeID const& nodeToSurvey)
     if (mPeersToSurvey.count(nodeToSurvey) != 0 ||
         nodeToSurvey == mApp.getConfig().NODE_SEED.getPublicKey())
     {
-#ifdef BUILD_TESTS
-        throw std::runtime_error("addPeerToBacklog failed: Peer is already in "
-                                 "the backlog, or peer is self.");
-#else
-        CLOG_ERROR(Overlay,
-                   "Tried to add node {} to survey backlog, but it is already "
-                   "queued or is the self node",
-                   KeyUtils::toStrKey(nodeToSurvey));
-        CLOG_ERROR(Overlay, "{}", REPORT_INTERNAL_BUG);
+        reportError("addPeerToBacklog failed: Peer is already in "
```

Comment:
This wasn't introduced by your changes, but I think simplest is to construct one error message that will be used in exceptions and logging (then the new helper function doesn't need to handle two different messages)